Assets/Scripts/Player/ActivatePowerup.cs

```diff
@@ -5,6 +5,8 @@
 public class ActivatePowerup : MonoBehaviour
 {
     public GameObject powerUpRing;
+    public bool hasPowerup = false;
```

Comment:
This can be private, doesn't seem like it'll be accessed by any other components